include/sparrow/arrow_interface/array_data_to_arrow_array_converters.hpp

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <type_traits>
+
+#include "sparrow/array/array_data.hpp"
+#include "sparrow/arrow_interface/arrow_array.hpp"
+
+namespace sparrow
+{
+    /**
+     * Convert an array_data vector to an ArrowArray vector.
+     *
+     * @param ads The array_data vector to convert.
+     * @return The converted children.
+     */
+    std::vector<arrow_array_shared_ptr> children_from_array_data_vec(std::vector<array_data>&& ads);
+    std::vector<arrow_array_shared_ptr> children_from_array_data_vec(const std::vector<array_data>& ads);
+
+    /**
+     * Convert array_data buffers to ArrowArray buffers.
+     *
+     * @tparam T A const reference or rvalue reference to an array_data.
+     * @param ad The array_data with the buffers convert.
+     * @return The converted buffers.
+     */
+    template <class T>
+        requires std::same_as<std::remove_cvref_t<T>, array_data>
+    std::vector<sparrow::buffer<uint8_t>> arrow_array_buffer_from_array_data(T&& ad);
+
+    /**
+     * Convert an array_data to an ArrowArray.
+     * 
+     * @tparam T A const reference or rvalue reference to an array_data.
+     * @param ad The array_data to convert.
+     * @return The converted ArrowArray. 
+     */
+    template <class T>
+        requires std::same_as<std::remove_cvref_t<T>, array_data>
+    arrow_array_unique_ptr from_array_data(T&& ad);
```

Comment:
naming: go the other way around: `to_output_type(input)`
same logic for all such conversion functions